activerecord/lib/active_record/enum.rb

```diff
@@ -128,13 +159,19 @@ def cast(value)
 
       def deserialize(value)
         return if value.nil?
+        return value if string_enum? && mapping.keys.include?(value)
         mapping.key(subtype.deserialize(value))
       end
 
       def serialize(value)
+        return value if string_enum? && mapping.keys.include?(value)
         mapping.fetch(value, value)
       end
 
+      def string_enum?
```

Comment:
What about making this helper method private?

Reply:
Yep. You're right! I'll fix it. Thanks)